src/content_mapblock.cpp

```diff
@@ -254,35 +254,57 @@ void MapblockMeshGenerator::drawCuboid(const aabb3f &box,
 	}
 }
 
+LightingPair convert_lighting(u16 light, u8 emissive_light)
+{
+	LightingPair result;
+	f32 day = (light & 0xff) / 255.0f;
+	f32 night = (light >> 8) / 255.0f;
+	result.artificial = std::min(1.0f, night + emissive_light * 0.01f);
+	// Since we don't know if the day light is sunlight or
+	// artificial light, assume it is artificial when the night
+	// light bank is also lit.
+	result.sunlight = std::max(0.0f, day - result.artificial);
+	return result;
+}
+
+video::SColor encode_light_and_color(const LightingSpec &light, video::SColor color)
+{
+	return video::SColor(light.sunlight_ratio,
+		light.average_light * color.getRed() / 256,
```

Comment:
what is better, store `u16` and divide by 256 (that is, simply use high-order byte) or store `u8` and divide by 255 (real division)?